src/Elastic.Apm/Api/IError.cs

```diff
@@ -1,30 +1,11 @@
 using System;
 using System.Collections.Generic;
+using Elastic.Apm.Model.Payload;
 
 namespace Elastic.Apm.Api
 {
 	public interface IError
```

Comment:
I exposed classes that are plain data transfer objects (DTOs) as they are, since the interface would be exactly the same as the implementations, and I added interface to classes that have some logic in them (like `StartSpan` on `Transaction`, etc..) - (`Error` is a little bit of an exception, that doesn't have too much logic, but the next point will give the reason for the `IError` interface). This is mainly because we agreed that we expose the server DTOs as they are, so by this agreement the interface and the implementation would always have exactly the same properties. The other idea is that I added interface for every event that we send to the server: if you look at `IPayloadSender` you see that it accepts the 3 interfaces I defined. If we need for example noop versions of the events later, we can add those easily.